servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/FailurePercentageXdsOutlierDetector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+
+import static io.servicetalk.loadbalancer.OutlierDetectorConfig.enforcing;
+
+final class FailurePercentageXdsOutlierDetector implements XdsOutlierDetector {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FailurePercentageXdsOutlierDetector.class);
+
+    // We use a sentinel value to mark values as 'skipped' so we don't need to create a dynamically sized
+    // data structure for doubles which would require boxing.
+    private static final long NOT_EVALUATED = Long.MAX_VALUE;
+
+    public static final XdsOutlierDetector INSTANCE = new FailurePercentageXdsOutlierDetector();
+
+    private FailurePercentageXdsOutlierDetector() {
+    }
+
+    @Override
+    public void detectOutliers(OutlierDetectorConfig config, Collection<XdsHealthIndicator> indicators) {
+        final long[] failurePercentages = new long[indicators.size()];
+        int i = 0;
+        int enoughVolumeHosts = 0;
+        int alreadyEjectedHosts = 0;
+        for (XdsHealthIndicator indicator : indicators) {
+            if (!indicator.isHealthy()) {
+                alreadyEjectedHosts++;
+            } else {
+                long successes = indicator.getSuccesses();
+                long failures = indicator.getFailures();
+                long totalRequests = successes + failures;
+                if (totalRequests >= config.failurePercentageRequestVolume()) {
```

Comment:
is it just me or the name is confusing? specifically the "percentage" part.

Reply:
I agree the names could be improved. Right now they follow the xDS lingo so I tried keeping with that but if we want to break from that and make the names a bit better we should change a few others as well.